js/coinsbit.js

```diff
@@ -0,0 +1,708 @@
+            //         start: 1640000000,
+            //         end: 1640001234,
+            //         interval: 60,
+            //         kline: [
+            //             {
+            //                 time: 1640000000,
+            //                 open: "46979.18",
+            //                 close: "46893.88",
+            //                 highest: "46979.18",
+            //                 lowest: "46893.88",
+            //                 volume: "2.35947525",
+            //                 amount: "110745.7371631963",
+            //                 market: "BTC_USDT"
+            //             },
+            //             {
+            //                 time: 1640000060,
+            //                 ....
+        }
+        const result = this.safeValue (response, 'result', {});
+        const data = this.safeValue (result, 'kline', []);
+        return this.parseOHLCVs (data, market, timeframe, since, limit);
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        //
+        // the ordering in spot candles is OCHLV
+        //
+        //     {
+        //         time: '1640051700',
+        //         open: '46954.3',
+        //         close: '46932.07',
+        //         highest: '46960.75',
+        //         lowest: '46932.07',
+        //         volume: '1.6985718',
+        //         amount: '79753.7310157163',
+        //         market: 'BTC_USDT'
+        //     }
+        //
+        // the ordering in swap / contract candles is OHLCV
+        //
+        if (market === undefined) {
+            const marketId = this.safeString (ohlcv, 'market');
+            market = this.market (marketId, market);
+        }
+        return [
+            this.safeTimestamp (ohlcv, 'time'),
+            this.safeNumber (ohlcv, 'open'),
+            this.safeNumber (ohlcv, 'highest'),
+            this.safeNumber (ohlcv, 'lowest'),
+            this.safeNumber (ohlcv, 'close'),
+            this.safeNumber (ohlcv, 'volume'), // At this moment, we do not make any use of 'amount' property
+        ];
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        params = this.handleMarketTypeAndParams ('fetchOHLCV', undefined, params)[1];
+        const request = {
+            'market': market['id']
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default = 50, minimum = 50, maximum = 1000
+        }
+        const defaultMethod = this.safeString (this.options, 'fetchTradesMethod');
+        const method = this.safeString (params, 'method', defaultMethod);
+        let data = {};
+        if (market['spot']) {
+            if (method === 'spotPublicGetHistory') {
+                if (!('lastId' in params)) {
+                    throw new ArgumentsRequired (this.id + " fetchTrades() requires 'lastId' id in params");
```

Comment:
Does the exchange API actually require a lastId in this case?

Reply:
yes, it requires : /   https://coinsbit.io/api/v1/public/history?market=BTC_USDT&lastId=2&limit=333 
remove lastId and you'll see

Reply:
I wonder where the user is supposed to find out the lastId, if it queries the endpoint for the first time in his life?

Reply:
cutting-in-the-middle question, I don't have an answer :) Moreover, if you look other endpoints (or overall whole available endpoints & functionality) , it's not surprising.

Reply:
Found the answer with a bit of guessing: https://coinsbit.io/api/v1/public/history?market=BTC_USDT&lastId=0

Reply:
Let's default to 0.